### PR TITLE
Fixed raw `=`-sign, AVAfied and ES2015ified the tests

### PR DESCRIPTION
**Pull request by SamVerschueren**
Firefox (pre 40) seems to decode `%3D` to `=` (see #35). This PR fixes that issue.

I also removed mocha in favorite of AVA (off course). Maybe I should've done this in separate PR's?

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -18,8 +18,8 @@ exports.parse = function (str) {
 
 	return str.split('&').reduce(function (ret, param) {
 		var parts = param.replace(/\+/g, ' ').split('=');
-		var key = parts[0];
-		var val = parts[1];
+		var key = parts.shift();
+		var val = parts.length > 0 ? parts.join('=') : undefined;
```
Comment (sindresorhus) [inline index.js]: Can add a comment about why this is like that and a link to this PR? So I can easily revert in the future when the Firefox version doesn't matter.

**Comment by SamVerschueren** [comment]
Something like this?

**Comment by arthurvr** [comment]
Thanks for doing this!

> Maybe I should've done this in separate PR's?

Yeah. Not a big deal though.

**Comment by sindresorhus** [comment]
Excellent. Thanks again! :)

> Maybe I should've done this in separate PR's?

Yes, that would have been preferable, but doesn't matter.

**Comment by sindresorhus** [comment]
Tip: You can run XO and AVA with th `--init --unicorn` flags to get my custom behavior of using `*` versions ;)

**Comment by SamVerschueren** [comment]
Oh thanks for the tips (again). Learning everyday contributing for you guys ;). :beers:

**Comment by SamVerschueren** [comment]
And thanks for merging! Appreciate it!

**Pull request closed without merging**